lib/rspec/core/formatters.rb

```diff
@@ -249,5 +251,15 @@ def underscore(camel_cased_word)
       word.downcase!
       word
     end
+
+    def file_at(path)
```

Comment:
I've improved the wording, but not the implementation, the problem is `Pathname` which behaves like a string and is returned from Rails things (which is why I believe we test for it at all) but we can't assume it's loaded etc, the existing behaviour is if we can write to the possible stream, return it, otherwise assume it's a filename and open it. I'd rather preserve that behaviour.